uniffi/src/bindings/kotlin/templates/Helpers.kt

```diff
@@ -0,0 +1,28 @@
+// A handful of classes and functions to support the generated data structures.
+// This would be a good candidate for isolating in its own ffi-support lib.
+
+abstract class FFIObject {
+    internal val isDestroyed = AtomicBoolean(false)
+
+    open fun destroy() {
+        isDestroyed.set(true)
+    }
+
+    internal inline fun <R> safelyCall(block: () -> R) =
```

Comment:
Naming nit: things with "safe" in their name are a personal bugbear of mine because it's often not obvious what "safe" means and they regularly turn out to be less safe than advertised. It's probably not a big deal for an internal helper function like this, but still, is there a more descriptive name?

One possible suggestion: if we moved `handle` up to be a property of this parent class, you could make this something like `callWithHandle`. Not sure if that would make the call-side code in each method declaration messier though.